### DISCUSSION TITLE
Does kohya read images in order, or randomly, when training?

### DISCUSSION THREAD

**Discussion by TeKett** [kohya-ss/sd-scripts#1728] [Q&A]
Just had this thought, couldn't find the answer. Since it should have a difference when using batches / accumulation. Since it could either have two very similar images, or two completely different images. So as a side question, which one is better. Grouped or random? As in, grouped in a folder per tag/character/artist/theme, or have a single folder where images have been given a random number.

**Comment by kohya-ss** [comment]
There is no difference between the two, as the datasets are all combined and then shuffled.
